### PR TITLE
LibLine+Shell: Support HISTCONTROL and HISTFILE env vars / js: load and save history

### PR DESCRIPTION
**Pull request by linusg**
Details in commit messages. cc @alimpfard

### CODE REVIEW

File: Base/usr/share/man/man7/Shell-vars.md [hunk 1]
```diff
@@ -17,7 +17,17 @@ The value of this variable is used to join lists or split strings into lists, it
 
 2. History
 
-`HISTFILE` (environment)
+`HISTCONTROL` (environment)
+
+The value of this variable is used to determine which entries are kept in the Shell's history, both regarding the current active session and when writing the history to disk on exit. It is unset by default, meaning no entries will be excluded from history.
```
Comment (alimpfard) [inline Base/usr/share/man/man7/Shell-vars.md:22]: > it is unset by default

but it being unset isn't documented...

Comment (linusg) [reply Base/usr/share/man/man7/Shell-vars.md:22]: > meaning no entries will be excluded from history

isn't that enough documentation for this case? :smile:

Comment (alimpfard) [reply Base/usr/share/man/man7/Shell-vars.md:22]: I'd just move the "meaning no entries..." to the list below under 'unset'.
I read that part when reading through and it didn't register at all...

Comment (linusg) [reply Base/usr/share/man/man7/Shell-vars.md]: cool, should be more clear now.

Comment (alimpfard) [reply Base/usr/share/man/man7/Shell-vars.md]: It is, thanks!

**Comment by awesomekling** [comment]
Very cool!

**Pull request merged**